pkg/cmd/cli/cmd/login.go

```diff
@@ -0,0 +1,250 @@
+				if options.usernameProvided() {
+					glog.V(5).Infof("Checking existing authentication info for '%v'...\n", options.username)
+
+					for key, authInfo := range configStore.Config.AuthInfos {
+						glog.V(5).Infof("Checking %v...\n", key)
+
+						if key == options.username && len(authInfo.Token) > 0 { // only token for now
+							glog.V(5).Infof("Authentication exists for '%v', trying to use it...\n", options.username)
+
+							clientCfg.BearerToken = authInfo.Token
+							if userFullName, err := whoami(clientCfg); err == nil && userFullName == options.username {
+								err = config.UpdateConfigFile(userFullName, authInfo.Token, f.OpenShiftClientConfig.KClientConfig, configStore)
+								checkErr(err)
+								requiresNewLogin = false
+								fmt.Printf("Already logged into %v as '%v'\n", clientCfg.Host, userFullName)
+							} else {
+								glog.V(5).Infof("Token %v no longer valid for '%v', need to log in again\n", authInfo.Token, options.username)
+							}
+						}
+					}
+				}
+
+				// at this point we know we need to log in again
+				if requiresNewLogin {
+					clientCfg.BearerToken = ""
+					accessToken, err := tokencmd.RequestToken(clientCfg, os.Stdin, options.username, options.password)
+					checkErr(err)
+
+					if userFullName, err := whoami(clientCfg); err == nil {
+						err = config.UpdateConfigFile(userFullName, accessToken, f.OpenShiftClientConfig.KClientConfig, configStore)
+						checkErr(err)
+						fmt.Printf("Logged into %v as %v\n", clientCfg.Host, userFullName)
+					} else {
+						glog.Fatalf("%v\n", err)
+					}
+				}
+			}
+
+			// select a project to use
+			// if the user only has one project, use that one
+			// if user has multiple, take the first and print the others
+			// if --project were provided, use that one
+			oClient, err := client.New(clientCfg)
+			checkErr(err)
+			projects, err := oClient.Projects().List(labels.Everything(), labels.Everything())
+			checkErr(err)
+
+			if size := len(projects.Items); size > 0 {
+				if options.projectProvided() {
+					currentCtx := configStore.Config.Contexts[configStore.Config.CurrentContext]
+					for _, project := range projects.Items {
+						if project.Name == options.project {
+							currentCtx.Namespace = project.Name
+							config := configStore.Config
+							config.Contexts[config.CurrentContext] = currentCtx
+
+							if err = kclientcmd.WriteToFile(*configStore.Config, configStore.Path); err != nil {
+								glog.Fatalf("Error saving config to file: %v", err)
+							}
+						}
+					}
+
+				}
+				if size > 1 {
+					projectsNames := make([]string, size)
+					for i, project := range projects.Items {
+						projectsNames[i] = project.Name
+					}
+					fmt.Printf("Your projects are: %v. You can switch between them at any time using 'osc project <project-name>'.\n", strings.Join(projectsNames, ", ")) // TODO switch projects TODO parameterize cmd name
+				}
+				current, err := f.OpenShiftClientConfig.Namespace()
```

Comment:
I'm pretty sure that at this point `OpenShiftClientConfig.Namespace()` can return a namespace that the user doesn't have rights to see.  That would be really weird.  You should make sure that the user has rights to whatever `.Namespace` is giving back and if they don't, let them know.